rfcs/0166-nix-formatting.md

```diff
@@ -0,0 +1,1268 @@
+However, the most important or potentially controversial rules are included, as well as some general meta-rules.
+
+When deciding between two *equally good* options, currently prevalent formatting style in Nixpkgs should be followed.
+The emphasis here is on "equally good".
+We should not fear making radical changes to the current style if there are sufficient arguments in favor of it.
+
+*Bad code does not deserve good formatting.*
+
+## Detailed design
+
+### Standard Nix format
+
+The _standard Nix format_ defines the officially recommended way that Nix code should be formatted.
+
+The initial version of the standard Nix format is defined in a section towards the end:
+
+[Initial standard Nix format](#initial-standard-nix-format).
+
+Significant changes to the standard Nix format must go through another RFC.
+
+The latest version of the standard Nix format must be in a file on the main branch of the [official Nix formatter](#official-nix-formatter).
+
+### Nix formatter team
+
+The _Nix formatter team_ is established, it has the responsibility to
+- Maintain the [official Nix formatter](#official-nix-formatter)
+- Regularly [reformat Nixpkgs](#reformat-nixpkgs) with it
+
+See the linked sections for more information.
+
+Initially the team consists of these members:
+- @piegames (author of this RFC, shepherd of the original formatting RFC)
+- @infinisil (from Tweag, co-author of this RFC, shepherd of the original formatting RFC)
+- @tomberek (from Flox, shepherd of the original formatting RFC)
+- @0x4A6F (shepherd of the original formatting RFC)
+- @Sereja313 (from Serokell, original sponsors of nixfmt)
+- @dasJ (from Helsinki Systems, writes Nix)
+
+Team member updates are left for the team itself to decide.
+
+### Official Nix formatter
+
+The Nix formatter team is given the authority and responsibility of
+creating and maintaining the _official Nix formatter_ implementation.
+This is a repository in the NixOS GitHub organisation.
+The repository will initially be based on [this nixfmt pull request](https://github.com/serokell/nixfmt/pull/118).
+This pull request has been developed along with this RFC and is already reasonably close to the proposed initial standard format.
+
+Any release of the official Nix formatter must conform to the latest version of the [standard Nix format](#standard-nix-format).
+
+The latest release of the official Nix formatter should support the Nix language syntax of the latest Nix release.
+The Nix formatter team should be consulted before the Nix language syntax is changed.
+
+### Reformat Nixpkgs
+
+For formatting Nixpkgs itself, a pinned release version of the official Nix formatter must be used.
+CI must generally enforce all files in Nixpkgs to be formatted with this version at all times.
+Automatically generated files may be handled differently, see [the next section](#automatically-generated-files).
+
```

Comment:
```suggestion
The Nixpkgs manual must also adhere to this Nix formatting standard and be formatted accordingly.

```

Just mentioning this, so it won't be forgotten when implementing.

Reply:
Put this in its own section: 4f99905ed2b4af7360afa4cd4046c74a68396afa